dashboard/test/models/course_test.rb

```diff
@@ -262,6 +262,64 @@ class SelectCourseScriptTests < ActiveSupport::TestCase
     end
   end
 
+  class WrongVersionWarningTests < ActiveSupport::TestCase
+    setup do
+      @csp_2017 = create(:course, name: 'csp-2017')
+      @csp1_2017 = create(:script, name: 'csp1-2017')
+      @csp2_2017 = create(:script, name: 'csp2-2017')
+      create :course_script, course: @csp_2017, script: @csp1_2017, position: 1
+      create :course_script, course: @csp_2017, script: @csp2_2017, position: 1
+
+      @csp_2018 = create(:course, name: 'csp-2018')
+      @csp1_2018 = create(:script, name: 'csp1-2018')
+      @csp2_2018 = create(:script, name: 'csp2-2018')
+      create :course_script, course: @csp_2018, script: @csp1_2018, position: 1
+      create :course_script, course: @csp_2018, script: @csp2_2018, position: 1
+
+      @csd = create(:course, name: 'csd')
+      @csd1 = create(:script, name: 'csd1')
+      create :course_script, course: @csd, script: @csd1, position: 1
+
+      @student = create :student
+    end
+
+    test 'validate test data' do
+      assert_equal 'csp', @csp_2017.assignment_family_name
+      assert_equal 'csp', @csp_2018.assignment_family_name
+
+      assert_equal 2, @csp_2017.default_scripts.count
+      assert_equal 2, @csp_2018.default_scripts.count
+      assert_equal 1, @csd.default_scripts.count
+    end
+
+    test 'student with no progress does not get warning' do
```

Comment:
naming of tests is tied to the warning view, but the tests are about the `wrong_version?` function on the model, which has no knowledge of the warning. I think these names should refer just to the model, or we should add tests to the controller to test the warning.